server/bin/pbench-restore-and-unpack-tarballs.py

```diff
@@ -0,0 +1,281 @@
+            cp_cmd = f"cp -a {tb} {tb}.md5 {ctrl_p}/".replace('"', '\\"').replace(
+                "'", "\\'"
+            )
+            print(f"\n{cp_cmd}", flush=True)
+            if not options.dry_run:
+                cp = subprocess.run(cp_cmd, shell=True, stderr=subprocess.STDOUT)
+                if cp.returncode != 0:
+                    print(
+                        f"FAILURE: cp command: '{cp_cmd}': {cp.returncode}, {cp.stdout!r}",
+                        file=sys.stderr,
+                    )
+                    return 1
+            # ... and run md5sum check against it
+            md5sum_cmd = f"md5sum --check {a_tb}.md5".replace('"', '\\"').replace(
+                "'", "\\'"
+            )
+            print(md5sum_cmd, flush=True)
+            if not options.dry_run:
+                # NOTE: we have to set the current working directory for the
+                # md5sum --check command to the controller directory since the
+                # contents of the .md5 file uses a relative file reference.
+                cp = subprocess.run(
+                    md5sum_cmd, shell=True, stderr=subprocess.STDOUT, cwd=str(ctrl_p)
+                )
+                if cp.returncode != 0:
+                    print(
+                        f"FAILURE: md5sum command: '{md5sum_cmd}': {cp.returncode}, {cp.stdout!r}",
+                        file=sys.stderr,
+                    )
+                    return 1
+
+            # Create the symlink recording the tar ball is already backed up
+            backed_up_dir = ctrl_p / "BACKED-UP"
+            if not options.dry_run:
+                backed_up_dir.mkdir(exist_ok=True)
+            backed_up = backed_up_dir / tb.name
+            print(f"ln -s {a_tb} {backed_up}", flush=True)
+            if not options.dry_run:
+                backed_up.symlink_to(a_tb)
+
+            # Create the symlink requesting the tar ball be unpacked
+            to_unpack_dir = ctrl_p / "TO-RE-UNPACK"
+            if not options.dry_run:
+                to_unpack_dir.mkdir(exist_ok=True)
+            to_unpack = to_unpack_dir / tb.name
+            print(f"ln -s {a_tb} {to_unpack}", flush=True)
+            if not options.dry_run:
+                to_unpack.symlink_to(a_tb)
+
+            tbs_restored += 1
+
+            restored_fmt = (
+                "\nFile {:d} of {:d} ({:d} restored,"
+                " {:d} existing, so far), restored: {}\n"
+            )
+            print(
+                restored_fmt.format(
+                    tbs_restored + tbs_existing,
+                    tbs_cnt,
+                    tbs_restored,
+                    tbs_existing,
+                    a_tb,
+                ),
+                flush=True,
+            )
+        except Exception as exc:
+            print(
+                f"Exception while processing ({tb_dt!r}, {ctrl!r}, {tb!r}): {exc}",
+                file=sys.stderr,
+            )
+            raise
```

Comment:
Are you sure that you want to do a `raise` here?  There is no surrounding `try` scope, so the exception will be unhandled.  This is probably OK on Linux, but it's not good on other platforms, and it's not good style in general.  I would think that the message alone would suffice, so you could replace the `raise` with a `return 1`.  (If you really want the traceback displayed, you can request that explicitly before exiting.)

Reply:
This is run at the command line by an admin, not in a cron job, and only on Linux.  Not sure about style, but this code is a stop-gap of sorts for the `0.69` release, so I'd rather have the exception for the time being.